src/main/java/com/github/retro_game/retro_game/entity/UserPasswordResetTokenKey.java

```diff
@@ -0,0 +1,42 @@
+package com.github.retro_game.retro_game.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.Embeddable;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import java.io.Serializable;
+import java.util.Objects;
+
+@Embeddable
+@NoArgsConstructor
+@AllArgsConstructor
+public class UserPasswordResetTokenKey implements Serializable {
+
+  @JoinColumn(name = "user_id", nullable = false, unique = true, updatable = false)
+  @OneToOne(fetch = FetchType.LAZY)
+  private User user;
+
+  public User getUser() {
+    return user;
+  }
+
+  public void setUser(User user) {
+    this.user = user;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    UserPasswordResetTokenKey that = (UserPasswordResetTokenKey) o;
+    return user.equals(that.user);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(user);
+  }
```

Comment:
`equals()` and `hashCode()` can also be replaced by `@EqualsAndHashCode` on the class, like in `DebrisFieldKey.java`.